extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchAggregatorFactory.java

```diff
@@ -62,6 +63,31 @@ public abstract class HllSketchAggregatorFactory extends AggregatorFactory
   private final boolean shouldFinalize;
   private final boolean round;
 
+  HllSketchAggregatorFactory(
+      final String name,
+      final String fieldName,
+      @Nullable final Integer lgK,
+      @Nullable final String tgtHllType,
+      @Nullable final StringEncoding stringEncoding,
+      final Boolean shouldFinalize,
+      final boolean round,
+      final int maxLgK
+  )
+  {
+    this(name, fieldName, lgK, tgtHllType, stringEncoding, shouldFinalize, round);
+
+    if (this.lgK > maxLgK) {
+      throw DruidException.forPersona(DruidException.Persona.USER)
+                          .ofCategory(DruidException.Category.INVALID_INPUT)
+                          .build(
+                              "LgK value [%s] for HLL sketch cannot be greater than [%s]. Reduce the lgK value"
```

Comment:
Please mention the runtime property name in the exception message.

Reply:
Added